packages/react/src/components/date-picker.tsx

```diff
@@ -52,6 +53,32 @@ const DatePicker: FC<DatePickerProps> = ({
       locale: _locale,
     });
 
+  const isMobile = useMediaQuery("(max-width: 640px)");
+
+  if (isMobile)
```

Comment:
Rather than using javascript to determine the screen size, use CSS instead to toggle which to show / hide

here's an example
```tsx
<>
  <button className="block lg:hidden">A</button> // Display on mobile
  <button className="hidden lg:block">B</button> // Display on tablet-desktop
<>
```